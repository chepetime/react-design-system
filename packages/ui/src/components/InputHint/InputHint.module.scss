@use "./../../styles" as styles;

:root {
  --inputhint-background-color: #{styles.$pink-500};
}

:global .dark {
  --inputhint-background-color: #{styles.$indigo-500};
}

.InputHint {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--inputhint-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
