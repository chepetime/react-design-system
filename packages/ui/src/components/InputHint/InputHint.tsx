import React from "react";
import classnames from "classnames";
import css from "./InputHint.module.scss";

export interface InputHintProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const InputHint = React.forwardRef<HTMLDivElement, InputHintProps>(
  (props: InputHintProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.InputHint)} ref={ref}>
        {props.children}
      </div>
    );
  }
);

InputHint.displayName = "InputHint";
