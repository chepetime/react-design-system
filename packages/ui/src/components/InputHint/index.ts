export { InputHint } from "./InputHint";
export type { InputHintProps } from "./InputHint";