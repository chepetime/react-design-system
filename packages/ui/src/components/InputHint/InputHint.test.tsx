import React from "react";
import { render } from "@testing-library/react";
import { InputHint } from "./InputHint";

describe("InputHint", () => {
  test("renders the InputHint component", () => {
    render(<InputHint>Field is required</InputHint>);
  });
});
