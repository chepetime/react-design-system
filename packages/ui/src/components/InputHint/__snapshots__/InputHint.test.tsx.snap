// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputHint matches snapshot 1`] = `
<div>
  <div
    class="InputHint"
  />
</div>
`;
